src/Service/CommentCRUDService.php

```diff
@@ -14,8 +18,39 @@ public function __construct(CommentRepository $commentRepository)
         $this->commentRepository = $commentRepository;
     }
 
-    public function saveComment(Comment $comment): void
-    {   
-        $this->commentRepository->add($comment);
+    /**
+     * @paramter Comment[] $posts
+     * @return Comment[] 
+     */ 
+    public function getAllCommentsByPosts(array $posts): array {
+        return $this->commentRepository->findAllCommentsByPosts($posts);
+    }
+
+    public function assignPost(Comment $comment, Post $post): void {
+        $comment->setPost($post);
+    }
+
+    public function assignDog(Comment $comment, Dog $dog): void {
+        $comment->setDog($dog);
+    }
+
+    public function handleAddComment(Comment $comment, Post $post, Dog $dog, FormInterface $form): bool
+    {
+        // is form posted in POST HTTP Response and is the form valid
+        if ($form->isSubmitted() && $form->isValid()) {
+
+            $this->assignPost($comment, $post);
+            $this->assignDog($comment, $dog);
+            $this->commentRepository->add($comment);
+
+            Log::info('CommentCRUDService@handleAddComment has added a comment', ['comment' => $comment, 'post' => $post, 'dog' => $dog]);
+
+
+            return true;
+        }
+
+        Log::info('CommentCRUDService@handleAddComment failed to add a comment', ['comment' => $comment, 'post' => $post, 'dog' => $dog]);
+
+        return false;
```

Comment:
This is rock solid code. Well done!!!

It may seem like overkill to have those one-line methods, but they allow you to extend what happens behind the scenes so much easier.